src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -110,6 +110,28 @@ class PythonSetup(Subsystem):
             help=(
                 "Set to true to enable the multiple resolves mechanism. See "
                 "`[python].resolves` for an explanation of this feature.\n\n"
+                "Warning: you may have issues with generating valid lockfiles via the "
+                "`generate-lockfiles` goal. Categorically, the resolves feature will not work if "
+                "you need to use VCS (Git) requirements and local requirements, along with "
+                "`[python-repos]` for custom indexes/cheeseshops. Several users have also had "
+                "issues with how Poetry's lockfile generation handles environment markers for "
+                "transitive dependencies; certain dependencies end up with nonsensical "
+                "environment markers which cause the dependency to not be installed, then "
+                "for Pants/Pex to complain the dependency is missing, even though it's in the "
+                "lockfile. The workaround is to manually create a `python_requirement` target for "
+                "the problematic transitive dependencies so that they are seen as direct "
+                "requirements, rather than transitive, then to regenerate the lockfile with "
+                "`generate-lockfiles`. We are working to fix these issues by using PEX & pip to "
+                "generate the lockfile.\n\n"
+                "Outside of the above issues with lockfile generation, we believe this "
```

Comment:
I don't understand how adding a warning also adds a sales pitch. I'd have thought the sales pitch would have been in the original help string then removed or amended by this warning.